lisa/tools/lspci.py

```diff
@@ -158,18 +185,52 @@ def get_devices_by_type(
             raise LisaException(
                 f"pci_type '{device_type}' is not supported to be searched."
             )
-        class_names = DEVICE_TYPE_DICT[device_type.upper()]
         devices_list = self.get_devices(force_run)
-        device_type_list = [x for x in devices_list if x.device_class in class_names]
+        device_type_list = []
+        for device in devices_list:
+            if device.device_id in DEVICE_ID_DICT[device_type.upper()]:
+                device_type_list.append(device)
+
         return device_type_list
 
+    @retry(KeyError, tries=10, delay=10)
```

Comment:
Add code comments above to explain the retry, and update `delay` to 1 or 2 seconds, so it can run faster.

Reply:
Added comments about retry.